### DISCUSSION TITLE
Why do we need two json libs: `System.Text.Json` and `System.Text.Json.Nodes`?

### DISCUSSION THREAD

**Discussion by B1Z0N** [dotnet/runtime#67033] [Q&A]
I see that the first is readonly and the second is able to create json from scratch, but then why shouldn't I just always use  the second(nodes)? 

My assumption: performance. Something else?

**Comment by gfoidl** [comment]
The first also allows for serialization/deserialization (in a efficient way). Also the source generators -- which generate code that aids for (de)serialization are for the first one.

With the first you can also write JSON, but not really in object-model way as with the latter. With the first you write to a json-writer directly, while the latter builds the json in a object-hierarchy in memory.

So the two target different areas, which one you use depends on the goals you want to achieve. It's another tool in your toolkit 😃.
